de.metas.marketing/base/src/main/java/de/metas/marketing/base/interceptor/MKTG_ContactPerson.java

```diff
@@ -75,4 +97,45 @@ public void deleteCampaignContactPersonAdUserId(@NonNull final I_MKTG_ContactPer
 				.addEqualsFilter(I_MKTG_Campaign_ContactPerson.COLUMN_MKTG_ContactPerson_ID, contactPerson.getMKTG_ContactPerson_ID())
 				.create();
 	}
+
+	@CalloutMethod(columnNames = { I_MKTG_ContactPerson.COLUMNNAME_EMail })
+	public void onChangeEmail(final I_MKTG_ContactPerson contactPersonRecord, final ICalloutField field)
```

Comment:
please extract this BL into a nice service, this method should just contain the code needed to set up the paramter(s) 
(get from the record classes like I_MKTG_ContactPerson to domain classes like ContactPerson )
and invoke the service method

Reply:
done